### PR TITLE
[FEATURE #1547] Support broker level entry filter for consumer

### PR DESCRIPTION
**Pull request by lwclover**
Fixes FEATURE #1547

### Motivation
KOP can support simple message filtering.

### Modifications
When consumer is sending fetch request, before the result backs to the consumer, filter with the broker level filters

Related components involved:
KafkaCommandDecod.handleFetchRequest
MessageFetchContext.handleEntries
EntryFormatter and sub class

### CODE REVIEW

**Review by BewareMyPower — CHANGES_REQUESTED** [review]
Could you add a test to test a customized entry filter?

**Comment by lwclover** [comment]
> Could you add a test to test a customized entry filter?

OK

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/format/AbstractEntryFormatter.java [hunk 1]
```diff
@@ -126,4 +133,24 @@ protected static boolean isKafkaEntryFormat(final MessageMetadata messageMetadat
         return false;
     }
 
+    protected EntryFilter.FilterResult filterOnlyByMsgMetadata(MessageMetadata msgMetadata, Entry entry,
+                                                                    List<EntryFilterWithClassLoader> entryFilters) {
```
[the remark this answers is not in the document]
Comment (lwclover) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/format/AbstractEntryFormatter.java:137]: done

**Review by BewareMyPower — DISMISSED** [review]

**Review by Demogorgon314 — COMMENTED** [review]
```java
EntryFormatterFactory.create(kafkaConfig, pulsarService.getBrokerService().getEntryFilters());
```
Can we pass the entry filter through the EntryFormatter constructors?

**Comment by lwclover** [comment]
> Can we pass the entry filter through the EntryFormaatter constructors?

Good idea

**Review by BewareMyPower — APPROVED** [review]

**Review by Demogorgon314 — APPROVED** [review]

**Pull request merged**